src/wing/CCPACSWingCell.cpp

```diff
@@ -351,46 +348,549 @@ std::pair<double, double> CCPACSWingCell::computePositioningEtaXsi(const CCPACSW
+    }
+    return cutGeometry.shape;
+}
+
+bool CCPACSWingCell::IsPartOfCell(TopoDS_Face f)
+{
+    Update();
+
+    Bnd_Box bBox1, bBox2;
+    BRepBndLib::Add(cellSkinGeometry, bBox1);
+    TopoDS_Face face = TopoDS::Face(m_parent->GetParentElement()
+                                        ->GetStructure()
+                                        .GetWingStructureReference()
+                                        .GetWingComponentSegment()
+                                        .GetWing()
+                                        .GetTransformationMatrix()
+                                        .Inverted()
+                                        .Transform(f));
+    BRepBndLib::Add(face, bBox2);
+
+    if (bBox1.IsOut(bBox2)) {
+        return false;
+    }
+
+    double u_min = 0., u_max = 0., v_min = 0., v_max = 0.;
+    Handle(Geom_Surface) surf = BRep_Tool::Surface(face);
+    BRepTools::UVBounds(face, u_min, u_max, v_min, v_max);
+    gp_Pnt pTest = surf->Value(u_min + ((u_max - u_min) / 2), v_min + ((v_max - v_min) / 2));
+
+    gp_Pnt midPnt = (pC1.XYZ() + pC2.XYZ()) / 2;
+    gp_Vec vTest(midPnt, pTest);
+    gp_Ax1 a1Test(midPnt, vTest);
+
+    bool sparTest = false, plainTest = false;
+
+    if (m_positioningLeadingEdge.GetInputType() == CCPACSWingCellPositionChordwise::InputType::Spar) {
+        sparTest = CCPACSWingShell::SparSegmentsTest(cutPlaneLE.Axis(), pTest, sparShapeLE);
+    }
+    else {
+        plainTest = a1Test.Angle(cutPlaneLE.Axis()) < M_PI_2;
+    }
+
+    if (plainTest || sparTest) {
+        // create an Ax1 from the trailing edge plane origin to the midpoint of the current face
+        midPnt = (pC3.XYZ() + pC4.XYZ()) / 2;
+        vTest  = gp_Vec(midPnt, pTest);
+        a1Test = gp_Ax1(midPnt, vTest);
+
+        sparTest  = false;
+        plainTest = false;
+        if (m_positioningTrailingEdge.GetInputType() == CCPACSWingCellPositionChordwise::InputType::Spar) {
+            sparTest = CCPACSWingShell::SparSegmentsTest(cutPlaneTE.Axis(), pTest, sparShapeTE);
+        }
+        else {
+            plainTest = a1Test.Angle(cutPlaneTE.Axis()) < M_PI_2;
+        }
+
+        if (plainTest || sparTest) {
+            // create an Ax1 from the inner border plane origin to the midpoint of the current face
+            midPnt = (pC1.XYZ() + pC3.XYZ()) / 2;
+            vTest  = gp_Vec(midPnt, pTest);
+            a1Test = gp_Ax1(midPnt, vTest);
+            if (a1Test.Angle(cutPlaneIB.Axis()) < M_PI_2) {
+                // create an Ax1 from the outer border plane origin to the midpoint of the current face
+                midPnt = (pC2.XYZ() + pC4.XYZ()) / 2;
+                vTest  = gp_Vec(midPnt, pTest);
+                a1Test = gp_Ax1(midPnt, vTest);
+                if (a1Test.Angle(cutPlaneOB.Axis()) < M_PI_2) {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+bool CCPACSWingCell::IsPartOfCell(TopoDS_Edge e)
```

Comment:
looks very similar to the version with TopoDS_Face. I am sure we can merge something here